addons/mail/static/src/models/thread_view/thread_viewer.js

```diff
@@ -219,6 +219,12 @@ function factory(dependencies) {
                 'discussHasThreadView',
             ],
         }),
+        /**
+         * Determines the selected `mail.message`.
+         */
+        selectedMessage: many2one('mail.message', {
+            related: 'discuss.replyingToMessage',
+        }),
```

Comment:
Should be the opposite: `Discuss.replyingToMessage` should rely on `ThreadViewer.selectedMessage`, not the other way around.

Reply:
Maybe, but it's not as simple as that, lot of things should be changed to reverse that. And I think it's kind of strange that discuss component change threadviewer selection instead of something related to discuss model..

Reply:
It's alright discuss => thread viewer because discuss is a kind of thread viewer. thread viewer => discuss feels wrong.
Current PR is better than before, so considering improving that takes more effort, I'm good merging this in the meantime.